buildSrc/src/main/groovy/airbyte-docker-convention.gradle

```diff
@@ -0,0 +1,90 @@
+import io.airbyte.gradle.GenerateConnectorBuildArgs
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.tasks.Copy
+
+class AirbyteDockerConventionPlugin implements Plugin<Project> {
+    private static final String SHARED_DOCKERFILE = 'docker-images/Dockerfile.java-connector-non-airbyte-ci'
+
+    @Override
+    void apply(Project project) {
+        /*
+            We deliberately avoid importing classes from the published Airbyte Docker
+            plugin.  That keeps buildSrc free of a compile‑time dependency on a specific
+            plugin version and lets this convention plugin compile before external
+            plugins are resolved.
+
+            Because we don’t have the plugin’s types on the class‑path, we access
+            task properties dynamically:
+
+            if (task.hasProperty('dockerfile')) { … }
+
+            This adds a few guard lines, but Groovy is dynamic anyway and the extra
+            resilience means the code continues to work even if a future Airbyte plugin
+            version renames or drops those properties.
+         */
+
+        project.plugins.withId('io.airbyte.gradle.docker') {
+            /* ---------- 0. default imageName = project.name -------------- */
+            project.extensions.findByName('airbyte')?.docker?.with {
+                if (!imageName.present) {
+                    imageName.convention(project.name)
+                }
+            }
+
+            /* ---------- 1. build‑args generator ------------------------- */
+            def genProps = project.tasks.register('generateConnectorDockerBuildArgs',
```

Comment:
Maybe rename to `dockerGenerateConnectorBuildArgs`?  In the base plugin I tried to have all the docker specific tasks start with `docker`.